com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs

```diff
@@ -48,48 +61,74 @@ public StackingSensor(ISensor wrapped, int numStackedObservations)
 
             m_Name = $"StackingSensor_size{numStackedObservations}_{wrapped.GetName()}";
 
-            if (wrapped.GetCompressionType() != SensorCompressionType.None)
-            {
-                throw new UnityAgentsException("StackingSensor doesn't support compressed observations.'");
-            }
-
-            var shape = wrapped.GetObservationShape();
-            if (shape.Length != 1)
-            {
-                throw new UnityAgentsException("Only 1-D observations are supported by StackingSensor");
-            }
-            m_Shape = new int[shape.Length];
+            m_WrappedShape = wrapped.GetObservationShape();
+            m_Shape = new int[m_WrappedShape.Length];
 
             m_UnstackedObservationSize = wrapped.ObservationSize();
-            for (int d = 0; d < shape.Length; d++)
+            for (int d = 0; d < m_WrappedShape.Length; d++)
             {
-                m_Shape[d] = shape[d];
+                m_Shape[d] = m_WrappedShape[d];
             }
 
             // TODO support arbitrary stacking dimension
-            m_Shape[0] *= numStackedObservations;
+            m_Shape[m_Shape.Length - 1] *= numStackedObservations;
+
+            // Initialize uncompressed buffer anyway in case python trainer does not
+            // support the compression mapping and has to fall back to uncompressed obs.
             m_StackedObservations = new float[numStackedObservations][];
             for (var i = 0; i < numStackedObservations; i++)
             {
                 m_StackedObservations[i] = new float[m_UnstackedObservationSize];
             }
+
+            if (m_WrappedSensor.GetCompressionType() != SensorCompressionType.None)
+            {
+                m_StackedCompressedObservations = new byte[numStackedObservations][];
+                m_EmptyCompressedObservation = CreateEmptyPNG();
+                for (var i = 0; i < numStackedObservations; i++)
+                {
+                    m_StackedCompressedObservations[i] = m_EmptyCompressedObservation;
+                }
+                m_CompressionMapping = ConstructStackedCompressedChannelMapping(wrapped);
+                m_tensorShape = new TensorShape(0, m_WrappedShape[0], m_WrappedShape[1], m_WrappedShape[2]);
```

Comment:
This needs to be set depending on `m_Shape.Length` - you can have uncompressed visual observations.